File: benchmarks/xlang/src/collatz.js
function steps(n) {
  let c = 0;
  while (n !== 1) { n = n % 2 === 0 ? n / 2 : 3 * n + 1; c++; }
  return c;
}
const t0 = performance.now();
let total = 0;
for (let i = 1; i <= 300000; i++) total += steps(i);
console.log(`CHECK ${total}`);
console.log(`MS ${Math.round(performance.now() - t0)}`);

File: benchmarks/xlang/src/collatz.rs
fn steps(mut n: i64) -> i64 {
    let mut c = 0;
    while n != 1 {
        n = if n % 2 == 0 { n / 2 } else { 3 * n + 1 };
        c += 1;
    }
    c
}
fn main() {
    let t0 = std::time::Instant::now();
    let mut total: i64 = 0;
    for i in 1..=300000 { total += steps(i); }
    println!("CHECK {}", total);
    println!("MS {}", t0.elapsed().as_millis());
}
